packages/gatsby-source-drupal/src/__tests__/index.js

```diff
@@ -34,7 +34,18 @@ describe(`gatsby-source-drupal`, () => {
   const { objectContaining } = expect
   const actions = {
     createNode: jest.fn(node => (nodes[node.id] = node)),
+    setPluginStatus: jest.fn(),
+    touchNode: jest.fn(),
   }
+  const getNodes = jest.fn(() => {
+    return {
+      filter: jest.fn(() => {
+        return {
+          forEach: jest.fn(() => nodes),
```

Comment:
This looks weird. `forEach` would never return anything.

Reply:
This doesn't actually need to return anything, I just needed it to exist so the `touchNode` code wouldn't cause an error during testing.

```
const existingNodes = getNodes().filter(
  n => n.internal.owner === `gatsby-source-drupal`
)
existingNodes.forEach(n => touchNode({ nodeId: n.id }))
```

I would be open for other ideas on how to do this, however, you have some other ideas on how to rewrite that code, so it will likely change anyway.